doc/developer-guide/generic_type.md

```diff
@@ -0,0 +1,127 @@
+# Generic Type
+
+## Generic Class
+
+### with type arguments
+``` TypeScript
+class A<T> {
+    _x: T;
+
+    constructor(x: T) {
+        this._x = x;
+    }
+    bar<T>(a: T) {
+        console.log('bar: ' + a);
+    }
+}
+```
+
+#### Creation of class type
+
+![](./img/generic_class_type_create.jpg)
+
+The steps to create a generic class type with type arguments are as follows:
+- create class scope  
```

Comment:
I think that the pipeline of creating generic class is same as creating normal class, what needed to be recorded is what's the difference of `TypeParameter` and other normal type.

Eg, how to store the `T`, and how to ensure that all `T` in the generic class is the same one.